### PR TITLE
add `resolve(id)` to BuildStep

### PR DESCRIPTION
**Pull request by jakemac53**
This adds essentially the `code_transformers` package logic directly to this package. It is up in the air still if this is really a good idea or not though ;).

cc @munificent @nex3 

closes https://github.com/dart-lang/build/issues/24

### CODE REVIEW

File: lib/src/analyzer/resolver.dart [hunk 1]
```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2016, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+import 'dart:async';
+
+import 'package:analyzer/src/generated/ast.dart' show Expression;
```
Comment (kegluneq) [inline lib/src/analyzer/resolver.dart]: Use the "visible" import `package:analyzer/analyzer.dart` instead

File: lib/src/builder/build_step_impl.dart [hunk 2]
```diff
@@ -5,16 +5,24 @@ import 'dart:async';
...
 import '../asset/writer.dart';
+import '../util/barback.dart';
 import 'build_step.dart';
 import 'exceptions.dart';
 
 /// A single step in the build processes. This represents a single input and
 /// its expected and real outputs. It also handles tracking of dependencies.
 class BuildStepImpl implements BuildStep {
+  static code_transformers.Resolvers _resolvers =
+      new code_transformers.Resolvers.fromMock(
```
Comment (kegluneq) [inline lib/src/builder/build_step_impl.dart]: I don't really understand `fromMock` - let's discuss offline.

File: lib/src/util/barback.dart [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        var message = new StringBuffer();
+        message.write('$level: ');
+        if (asset != null) {
+          message.write('From $asset');
+          if (span != null) {
+            message.write('()$span)');
+          }
+          message.write(': ');
+        }
+        message.write(message);
```
Comment (kegluneq) [inline lib/src/util/barback.dart]: We must go deeper (two `message`s in scope here.)

Comment (jakemac53) [inline lib/src/util/barback.dart]: hahaha good catch :)

File: lib/src/builder/build_step_impl.dart (hunk 2 above)
Comment (jakemac53) [inline lib/src/builder/build_step_impl.dart]: Will switch this to use the actual dart sdk until we figure out how we want to make it configurable.

File: lib/src/analyzer/resolver.dart (hunk 1 above)
Comment (jakemac53) [inline lib/src/analyzer/resolver.dart]: done

**Comment by jakemac53** [comment]
updated

**Comment by kegluneq** [comment]
For future reviews, please push a commit with just review changes overtop the original one. You can always merge them before committing.

LGTM

**Pull request merged**